File: 프로그래머스/기초 트레이닝 - Level 0/전국대회선발고사-solve.java
class Solution {
    public int solution(int[] rank, boolean[] attendance) {
        int[] top3 = new int[4]; 
        int targetRank = 1; 
        int candidateRank = 1; 
        
        while (targetRank <= 3) {
            for (int i = 0; i < rank.length; i++) {
                if (rank[i] == candidateRank) {
                    if (attendance[i]) {
                        top3[targetRank] = i;
                        targetRank++;
                    }
                }
            }
            candidateRank++;
        }
        
        return 10000 * top3[1] + 100 * top3[2] + top3[3]; 
    }
}

File: 프로그래머스/기초 트레이닝 - Level 0/전국대회선발고사-solve.py
"""
0번부터 n - 1번까지 n명의 학생 중 3명 선발
참여가능한 학생 중 등수가 높은 3명 선발 
rank: 각 학생들 선발고사 등수 정수 배열 
attendance: 참여 가능 여부 boolean 배열 
return : 선발된 학생 번호 등수 높은 순서대로 a, b, c에서 10000 * a + 100 * b + c 
"""

def solution(rank, attendance):
    result = sorted([(rank[i], i) for i in range(len(rank)) if attendance[i]])
    return 10000 * result[0][1] + 100 * result[1][1] + result[2][1]